src/credentials/FabricTable.cpp

```diff
@@ -117,7 +117,10 @@ CHIP_ERROR FabricInfo::CommitToStorage(PersistentStorageDelegate * storage)
 
         ReturnErrorOnFailure(writer.Put(kOpKeyVersionTag, kOpKeyVersion));
 
-        ReturnErrorOnFailure(writer.Put(kOpKeyDataTag, ByteSpan(serializedOpKey.Bytes(), serializedOpKey.Length())));
+        if (!mHasExternallyOwnedOperationalKey)
+        {
+            ReturnErrorOnFailure(writer.Put(kOpKeyDataTag, ByteSpan(serializedOpKey.Bytes(), serializedOpKey.Length())));
```

Comment:
But now `FabricInfo::LoadFromStorage` will fail for this fabric index.  That function needs changes to deal with a missing `kOpKeyDataTag`, no?

Reply:
I'd think that we should:

1. Assume the FabricInfo is externally owned if `kOpKeyDataTag` is not encountered in `LoadFromStorage`.
2. There is now a period of time between `LoadFromStorage` till when `InitControllerNOCChain` is called on the controller instance associated with the respective `FabricInfo` that the fabric info instance is actually _invalid_. I think that should be caught in `GetOperationalKey`, which is the getter that most external consumers use to get at the key. Presumably, that should just return `mOperationalKey` regardless of whether it is null or not if the flag is set to true?

Reply:
No, because it will never get called with anything in it from a controller that externally manages everything

Reply:
It will get called in all cases I believe, due to this call in `DeviceControllerFactory::InitSystemState` --     

```
ReturnErrorOnFailure(stateParams.fabricTable->Init(params.fabricIndependentStorage));
```

Reply:
@mrjerryjohns It will get called, but you could have it empty at the start everytime, in which case nothing gets loaded.

Reply:
Discussed this with @tcarmelveilleux.  What we should do is this: when loading a fabric table entry from storage, if it has an externally managed key (which we can either detect via "we did not have a tag for the key in this TLV" or we could store the boolean for "has externally managed key" explicitly") we should just set `mOperationalKey` to nullptr.  Then we modify `FabricInfo::GetOperationalKey` to just return `mOperationalKey` if `mHasExternallyOwnedOperationalKey` without trying to heap-allocate it there.

The callers of `GetOperationalKey()` all do null-checks already, so will correctly handle it being null (typically by doing things like erroring out of CASE session establishment, which is what we want anyway in that situation).